### PR TITLE
fix(manager/gitlabci): support alias with a path in component URL

### PR DESCRIPTION
**Pull request by marcovmun**
## Changes

Support aliases that include a path, because replacement was happening too late in the function.

## Context

Components with path like this were not supported:
```
include:
  - component: $COMPONENT_REGISTRY/a-project/a-component@1.0
```

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/gitlabci/extract.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ import { getGitlabDep, replaceReferenceTags } from './utils';
 const componentReferenceRegex = regEx(
   /(?<fqdn>[^/]+)\/(?<projectPath>.+)\/(?:.+)@(?<specificVersion>.+)/,
 );
+const componentFQDNRegex = regEx(/[^/]+/);
```
Comment (viceice) [inline lib/modules/manager/gitlabci/extract.ts:34]: I don't understand why we need this simple regex? what do you check with that? isn't a simple string check enough?

File: lib/modules/manager/gitlabci/common.ts [hunk 2]
```diff
@@ -37,5 +37,8 @@ export function isGitlabIncludeLocal(
 export function isGitlabIncludeComponent(
   include: GitlabInclude,
 ): include is GitlabIncludeComponent {
-  return !is.undefined((include as GitlabIncludeComponent).component);
+  return (
+    !is.undefined((include as GitlabIncludeComponent).component) &&
+    typeof (include as any).component === 'string'
+  );
```
Comment (rarkins) [inline lib/modules/manager/gitlabci/common.ts:43]: Is `is.string(.....)` an alternative here?

File: lib/modules/manager/gitlabci/extract.ts [hunk 3]
```diff
@@ -122,9 +122,16 @@ function extractDepFromIncludeComponent(
   includeComponent: GitlabIncludeComponent,
   registryAliases?: Record<string, string>,
 ): PackageDependency | null {
-  const componentReference = componentReferenceRegex.exec(
-    includeComponent.component,
-  )?.groups;
+  let componentPath = includeComponent.component;
+  const componentFQDN = componentPath?.substring(0, componentPath.indexOf('/'));
```
Comment (viceice) [inline lib/modules/manager/gitlabci/extract.ts:126]:
```suggestion
  const componentFQDN = componentPath.substring(0, componentPath.indexOf('/'));
```
I think component path can't be empty with your changed check above.

File: lib/modules/manager/gitlabci/common.ts (hunk 2 above)
Comment (viceice) [reply lib/modules/manager/gitlabci/common.ts:43]: no any cast.

File: lib/modules/manager/gitlabci/extract.ts [hunk 4]
```diff
@@ -122,9 +122,16 @@ function extractDepFromIncludeComponent(
   includeComponent: GitlabIncludeComponent,
   registryAliases?: Record<string, string>,
 ): PackageDependency | null {
-  const componentReference = componentReferenceRegex.exec(
-    includeComponent.component,
-  )?.groups;
+  let componentPath = includeComponent.component;
+  const componentFQDN = componentPath.substring(0, componentPath.indexOf('/'));
```
Comment (viceice) [inline lib/modules/manager/gitlabci/extract.ts:126]: why this substring? makes no sense to me. used should be able to alias more than the first part.

```suggestion
  const componentFQDN = componentPath;
```

File: lib/modules/manager/gitlabci/extract.ts [hunk 5]
```diff
@@ -122,9 +122,16 @@ function extractDepFromIncludeComponent(
...
+  if (componentFQDN && registryAliases?.[componentFQDN]) {
```
Comment (viceice) [inline lib/modules/manager/gitlabci/extract.ts:127]:
```suggestion
  if (registryAliases?.[componentFQDN]) {
```

**Review by viceice — COMMENTED** [review]
looks like we can drop the `componentFQDN` variable

File: lib/modules/manager/gitlabci/extract.ts (hunk 4 above)
Comment (marcovmun) [reply lib/modules/manager/gitlabci/extract.ts:126]: I don't get it, `componentPath` is always combination of a variable and a path to the specifiek component and version. The reason for the aliases to replace the first path of the string with the absolute URL.
Like in the test:
```
- component: $COMPONENT_REGISTRY/a-project/a-component@1.0
```
You want to get `$COMPONENT_REGISTRY` from the componentPath to check if variable can be found in the `registryAliases` array.

If you want to be more dynamic, then we need to loop through all aliases and check if alias can be found in the component path and replace that. But that was not how the old implementation was working, but can be added.

```
// dummy code
for (registryAliases as key => value} {
     componentPath = componentPath.replace(key, value)
}
```

But maybe I don't understand your suggestion.

**Review by viceice — APPROVED** [review]

**Pull request merged**